### DISCUSSION TITLE
Help uploading image & displaying

### DISCUSSION THREAD

**Discussion by RealityMasque** [Megabit/Blazorise#5019] [General]
I'm trying to make an online character sheet, & thought I could have character potrait section.  the user would upload an image, then it would be displayed in an image.  unfortunately, the type in the arguments after uploading a file for both the fileedit & filepicker is an IFileEntry.  I can read that into a memory stream, but everything I've tried thus far saving that to a file on my computer results in some kind of file that can't be opened & viewed on my machine, with something like paint.net .  am I missing something?

**Comment by stsrki** [comment]
The file should have the same `byte[]` content as an original. Can you open a new issue ticket with a code to reproduce it on our side?

**Comment by RealityMasque** [comment]
figured it out - I wasn't seeking to the beginning of the memory stream before saving to file.

**Comment by RealityMasque** [comment]
great - I spoke too soon.  I got it working on my work laptop, but when I put the same code (w/ destination save path changes) on my personal laptop, the "await file.WriteToStreamAsync( stream );" call doesn't worked.  do I need to run VS in admin mode?  I even tried making the dest folder full access to everyone, but the writing to the memorystream happens before writing to disk...

**Reply by David-Moreira** [reply]
Hello,

Can you show us your actual code? Is there an exception being thrown?

This should work : 

```
        var filePath = Path.Combine( _path, file.FileName );

        using var stream = new FileStream( filePath, FileMode.Create );
        await file.WriteToStreamAsync( stream );
```
OR

```
        var filePath = Path.Combine( _path, file.FileName );

        using var stream = new FileStream( filePath, FileMode.Create );
        await file.OpenReadStream( long.MaxValue ).CopyToAsync( stream );
```

![image](https://github.com/Megabit/Blazorise/assets/22283161/366a8b18-742c-4a1b-8369-1dd73c645d71)

**Notes:**
- Don't forget that with webassembly you can't write directly to disk
- Make sure that your call stack actually awaits for the file operation to complete
